### ISSUE TITLE
Set ROM inflector be the same as the Hanami app inflector

### ISSUE THREAD

**Issue by timriley** [hanami/hanami#1400]
In the persistence provider for my example app decafsucks, [we need to do the following](https://github.com/decafsucks/decafsucks/blob/5923add8c29f23e4259e7d43d69de3b57b12c737/config/providers/persistence.rb#L9-L10):

```ruby
    # TODO(Hanami): As part of built-in rom setup, configure ROM with app inflector
    silence_warnings { ROM::Inflector = Hanami.app["inflector"] }
```

ROM 5.x doesn't support a configurable inflector.

We're either going to need to hardcode a hack like this, or update ROM to allow this to be configurable, and then properly pass in the inflector.

**Comment by alassek** [comment]
Really should fix ROM, but this is more difficult due to the unfinished state 6.0 is in -- probably would need to fix it in 6 and backport to 5.3

**Comment by alassek** [comment]
I didn't realize [ROM 6 changed this](https://github.com/rom-rb/rom/pull/591) 🤔 

Backporting this looks like a _lot_ of work. Personally, I think the hacky fix for 5.3 is probably best and the conversation should really be about what is holding back 6.0 from adoption.

**Comment by timriley** [comment]
@alassek yeah, I agree. Let's follow the principle of "avoid modifying rom" and just do what's necessary to get our inflector set.
